lfsapi/client.go

```diff
@@ -159,15 +159,15 @@ func (c *Client) doWithRedirects(cli *http.Client, req *http.Request, via []*htt
 
 	requests := tools.MaxInt(0, retries) + 1
 	for i := 0; i < requests; i++ {
+		if seek, ok := req.Body.(io.Seeker); ok {
```

Comment:
I think that the performance impact is negligible, but could we avoid making this `Seek` call on the first run of the loop?